app/src/main/java/com/mashup/mobalmobal/ui/profile/presenter/ProfileViewModel.kt

```diff
@@ -3,52 +3,143 @@ package com.mashup.mobalmobal.ui.profile.presenter
     val toastSubject get() = _toastSubject
 
-    private val _loadingSubject: PublishSubject<Boolean> = PublishSubject.create()
-    val loadingSubject get() = _loadingSubject
+    val userNickSubject: Observable<String> get() = _userDtoSubject.map { it.nickname }
 
-    private val _profileSubject: BehaviorSubject<List<ProfileItem>> =
-        BehaviorSubject.createDefault(emptyList())
-    val profileSubject get() = _profileSubject
+    private val _backTriggerSubject: PublishSubject<Unit> = PublishSubject.create()
+    val backTriggerSubject get() = _backTriggerSubject
+
+    private val _userDtoSubject: BehaviorSubject<UserDto> = BehaviorSubject.create()
 
     init {
-        MobalSharedPreferencesImpl.getUserId()?.let { requestProfile(it) }
+        userRepository.fetchUser()
+            .subscribeOnIO()
+            .subscribeWithErrorLogger { userDto ->
+                userDto.data?.let { _userDtoSubject.onNext(it) }
+            }.addToDisposables()
     }
 
-    fun requestProfile(userId: String) {
-        profileRepository.getProfile(userId)
-            .toFlowable()
+    private val _donationsSubject: BehaviorSubject<List<ProfileItem>> =
+        BehaviorSubject.createDefault(emptyList())
+
+    init {
+        requestDonations()
             .subscribeOnIO()
-            .doOnSubscribe { _loadingSubject.onNext(true) }
-            .subscribeWithErrorLogger { response ->
-                with(response) {
-                    if (result != Constants.NETWORK_SUCCESS) {
-                        throw IOException(data.message)
-                    }
-                    _profileSubject.onNext(data.toProfileItems())
-                }
-                _loadingSubject.onNext(false)
+            .subscribeWithErrorLogger {
+                _donationsSubject.onNext(it)
+            }
+            .addToDisposables()
+    }
+
+    private val _itemSubject: BehaviorSubject<List<ProfileItem>> = BehaviorSubject.create()
+    val itemSubject: Observable<List<ProfileItem>> = _itemSubject.distinctUntilChanged()
+
+    init {
+        _userDtoSubject.combineLatest(_donationsSubject)
+            .distinctUntilChanged()
+            .map { listOf(it.first.toProfileItem()) + it.second }
+            .subscribeWithErrorLogger {
+                _itemSubject.onNext(it)
             }
             .addToDisposables()
     }
 
+    private fun requestDonations() =
+        requestMyCreateDonations()
+            .zipWith(requestMyDonated())
+            .zipWith(requestMyClosed())
+            .map {
+                val createDonations =
+                    it.first.first.toProfileItems(R.string.profile_header_donation_request)
+                val donatedDoantions =
+                    it.first.second.toProfileItems(R.string.profile_header_donated)
+                val closedDonations =
+                    it.second.toProfileItems(R.string.profile_header_donation_closed)
```

Comment:
```suggestion
            .map { (myCreateDonationsWithMyDonated, myClosed) ->
                val (myCreateDonations, myDonated) = myCreateDonationsWithMyDonated
                val createDonations = myCreateDonations.toProfileItems(R.string.profile_header_donation_request)
                val donatedDoantions = myDonated.toProfileItems(R.string.profile_header_donated)
                val closedDonations = myClosed.toProfileItems(R.string.profile_header_donation_closed)
```
it.frst.first 방식이 가독성 부분에서 좀 떨어지는것 같아요. 위의 방식 어떠세요?
만약 위에 작성한 function이 있으면 세 Single를 하나의 Triple로 내보내 주니 아래의 방식도 가능합니다.
```suggestion
            .map { (myCreateDonations, myDonated, myClosed) ->
                val createDonations = myCreateDonations.toProfileItems(R.string.profile_header_donation_request)
                val donatedDoantions = myDonated.toProfileItems(R.string.profile_header_donated)
                val closedDonations = myClosed.toProfileItems(R.string.profile_header_donation_closed)
```

Reply:
funibase 업데이트 되면 수정할게요~!

Reply:
업데이트 안될 것 같아요 ㅜㅜ 
그냥 위의걸로 가면 될 것 같아요!

Reply:
헛 네넵 ㅠ